File: ecbs-os/backend/src/ECBS.Application/ScreenData/DeploymentDocumentationData.cs
namespace ECBS.Application.ScreenData;

public sealed record DeploymentDocumentationData(
    IReadOnlyList<DocumentDataRow> DocumentRows,
    IReadOnlyList<AnalysisSummaryRow> SummaryRows,
    IReadOnlyList<AnalysisSummaryRow> MetadataRows,
    IReadOnlyList<AnalysisSummaryRow> FolderRows,
    IReadOnlyList<AnalysisSummaryRow> PermissionRows,
    IReadOnlyList<DocumentDataRow> ReviewRows,
    IReadOnlyList<DocumentDataRow> SearchRows,
    IReadOnlyList<DocumentDataRow> VersionRows,
    string ClientName,
    string DeploymentId,
    string Message,
    string ProjectName,
    string SiteName,
    string State,
    string Status,
    string UpdatedAt);

public sealed record DocumentDataRow(
    string Id,
    string Name,
    string Type,
    string Folder,
    string UploadedBy,
    string UploadedAt,
    string Status,
    string Size,
    string StorageUri);

File: ecbs-os/frontend/src/lib/deploymentDocumentationData.ts
import type { AnalysisSummaryRow } from "@/lib/analysisData";

export type DocumentDataRow = {
  folder: string;
  id: string;
  name: string;
  size: string;
  status: string;
  storageUri: string;
  type: string;
  uploadedAt: string;
  uploadedBy: string;
};

export type DeploymentDocumentationData = {
  clientName: string;
  deploymentId: string;
  documentRows: DocumentDataRow[];
  folderRows: AnalysisSummaryRow[];
  message: string;
  metadataRows: AnalysisSummaryRow[];
  permissionRows: AnalysisSummaryRow[];
  projectName: string;
  reviewRows: DocumentDataRow[];
  searchRows: DocumentDataRow[];
  siteName: string;
  state: string;
  status: string;
  summaryRows: AnalysisSummaryRow[];
  updatedAt: string;
  versionRows: DocumentDataRow[];
};

